notebooks/notebook_utils.py

```diff
@@ -0,0 +1,62 @@
+import re
```

Comment:
I found a way to store the data in the notebook without displaying it like scrapbook, but much simpler:
```
name = "metric"
metric = 1.3234324
metadata = {"notebook_utils": dict(name=name, data=True, display=False)}
data_json = {"application/notebook_utils.json+json": {"name": name, "data": metric, "encoder": "json"} } 
display(data_json, metadata=metadata, raw=True)
```
after executing the code above **and saving the notebook**, the notebook source is:
```
{
   "cell_type": "code",
   "execution_count": 43,
   "id": "338b85a4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/notebook_utils.json+json": {
       "data": 1.3234324,
       "encoder": "json",
       "name": "metric"
      }
     },
     "metadata": {
      "notebook_utils": {
       "data": true,
       "display": false,
       "name": "metric"
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "name = \"metric\"\n",
    "metric = 1.3234324\n",
    "metadata = {\"notebook_utils\": dict(name=name, data=True, display=False)}\n",
    "data_json = {\"application/notebook_utils.json+json\": {\"name\": name, \"data\": metric, \"encoder\": \"json\"} } \n",
    "display(data_json, metadata=metadata, raw=True)\n"
   ]
  },
```

We can also have multiple instances of this data storage. The data will be saved, but not displayed, and concatenated in the notebook json:
```
name = "metric10"
metric = 43523
metadata = {"notebook_utils": dict(name=name, data=True, display=False)}
data_json = {"application/notebook_utils.json+json": {"name": name, "data": metric, "encoder": "json"} } 
display(data_json, metadata=metadata, raw=True)


name = "metric20"
metric = "aaabbb"
metadata = {"notebook_utils": dict(name=name, data=True, display=False)}
data_json = {"application/notebook_utils.json+json": {"name": name, "data": metric, "encoder": "json"} } 
display(data_json, metadata=metadata, raw=True)

```

the notebook json is:
```
{
   "cell_type": "code",
   "execution_count": 42,
   "id": "6d751c8f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/notebook_utils.json+json": {
       "data": 43523,
       "encoder": "json",
       "name": "metric10"
      }
     },
     "metadata": {
      "notebook_utils": {
       "data": true,
       "display": false,
       "name": "metric10"
      }
     },
     "output_type": "display_data"
    },
    {
     "data": {
      "application/notebook_utils.json+json": {
       "data": "aaabbb",
       "encoder": "json",
       "name": "metric20"
      }
     },
     "metadata": {
      "notebook_utils": {
       "data": true,
       "display": false,
       "name": "metric20"
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "name = \"metric10\"\n",
    "metric = 43523\n",
    "metadata = {\"notebook_utils\": dict(name=name, data=True, display=False)}\n",
    "data_json = {\"application/notebook_utils.json+json\": {\"name\": name, \"data\": metric, \"encoder\": \"json\"} } \n",
    "display(data_json, metadata=metadata, raw=True)\n",
    "\n",
    "\n",
    "name = \"metric20\"\n",
    "metric = \"aaabbb\"\n",
    "metadata = {\"notebook_utils\": dict(name=name, data=True, display=False)}\n",
    "data_json = {\"application/notebook_utils.json+json\": {\"name\": name, \"data\": metric, \"encoder\": \"json\"} } \n",
    "display(data_json, metadata=metadata, raw=True)"
   ]
  }

```